src/motion_click_app.c

```diff
@@ -0,0 +1,156 @@
+/***************************************************************************************************
+ * Copyright (c) 2016, Imagination Technologies Limited and/or its affiliated group companies
+ * and/or licensors
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted
+ * provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+ *    and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other materials provided
+ *    with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to
+ *    endorse or promote products derived from this software without specific prior written
+ *    permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <poll.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include "gpio.h"
+
+#define INFINITE_TIMEOUT  -1
+
+static bool running = true;
+
+static void signal_handler(int sig)
+{
+    running = false;
+    signal(sig, SIG_IGN);
+}
+
+static void print_usage(const char *program)
+{
+    printf("Usage: %s [options]\n\n"
+        " -m  <arg>  mikroBUS number\n"
+        " -h         display this message\n\n",
+        program);
+}
+
+static int parse_cmd_opts(int argc, char *argv[], uint8_t *gpio)
+{
+    int opt;
+    uint8_t num;
+
+    opterr = 0;
+
+    while (1)
+    {
+        opt = getopt(argc, argv, "m:t:h");
+        if (opt == -1)
```

Comment:
change the code to get rid of this extra if - it will make the code far more readable